sdk/tables/azure-data-tables/tests/test_table_entity.py

```diff
@@ -1282,14 +1284,43 @@ def test_delete_entity_with_if_doesnt_match(self, tables_storage_account_name, t
             # Act
             with pytest.raises(HttpResponseError):
                 self.table.delete_entity(
-                    entity.PartitionKey, entity.RowKey,
+                    {"PartitionKey": entity.PartitionKey, "RowKey": entity.RowKey},
                     etag=u'W/"datetime\'2012-06-15T22%3A51%3A44.9662825Z\'"',
-                    match_condition=MatchConditions.IfNotModified)
+                    match_condition=MatchConditions.IfNotModified
+                )
 
             # Assert
         finally:
             self._tear_down()
 
+    @TablesPreparer()
+    def test_delete_entity_overloads(self, tables_storage_account_name, tables_primary_storage_account_key):
+        # Arrange
+        self._set_up(tables_storage_account_name, tables_primary_storage_account_key)
+        try:
+            entity, _ = self._insert_random_entity()
+
+            # Act
+            self.table.delete_entity(entity)
+
+            pk, rk = self._create_pk_rk("pk", "rk")
+            pk, rk = pk + u"2", rk + u"2"
+            entity2 = {
+                u"PartitionKey": pk,
+                u"RowKey": rk,
+                u"Value": 100
+            }
+            self.table.create_entity(entity2)
+
+            self.table.delete_entity(pk, rk)
```

Comment:
Can we try these both with kwargs as well?
```python
self.table.create_entity(entity=entity2)
self.table.delete_entity(partition_key=pk, row_key=rk)
```

Reply:
Taken care of, slight change to the logic of parsing args. Added as a second test.

Reply:
Confirmed it works for Py 2.7 too.